04-suspense-and-async-mode/README.md

```diff
@@ -0,0 +1,245 @@
+> [Source: Andrew Clark's twitter post](https://twitter.com/acdlite/status/977291318324948992)
+
+* Formerly known as "Async mode", but "Concurrent mode" name has been chosen eventually because it reflects more the ability to perform work on different priority levels
+* Allows React to interrupt a long-running render (for example, rendering a new feed story) to handle a high-priority event (for example, text input or hover)
+* Keeps your app responsive while rendering complex component trees
+* Allows Suspense to skip unnecessary loading states on fast connections (flashing spinners)
+* Concurrent mode is the future of React, but so far only in experimental state.
+* In Road map planned for Q2 2019
+* No documentation yet
+
+If you want to know more about how the concurrent mode is implemented, check out `fiber` and `algebraic effects`
+
+### Enabling The Concurrent Mode
+
+Instead of
+
+```javascript
+ReactDOM.render(<App />, document.getElementById('root'));
+```
+
+simply use:
+
+```javascript
+ReactDOM
+  .createRoot(document.getElementById('root'))
+  .render(<App />);
+```
+
+This will allow to use new features, e.g. maxDuration prop of Suspense - it defines the time in ms after which our fallback component will show up. This will avoid screen flickering issue which usually occurs on faster network where the loader shows up for few ms and then the data comes immediately.
+
+### Important Changes In React With Concurrent Mode
+
+As render can be called multiple times in Concurrent Mode, some methods are no longer safe to use. Wrap your application in `<React.StrictMode>` to reveal potential problems - [see more here](https://reactjs.org/docs/strict-mode.html#identifying-unsafe-lifecycles)
+
+Following lifecycle methods will be deprecated in future:
+
+* `componentWillMount`
+* `componentWillReceiveProps`
+* `componentWillUpdate`
+
+Two new lifecycle methods will be added as a replacement:
+
+* `getSnapshotBeforeUpdate`
+* `getDerivedStateFromProps`
+
+![Lifecycle Methods Overview](./images/concurrent_lifecycle_methods.jpeg)
+> [Source: Dan Abramov's twitter post](https://twitter.com/dan_abramov/status/981712092611989509/photo/1)
+
+***It's crucial to have your render methods/functions pure, without any side effects!***
+
+## Sources and Further Reading
+
+### Talks
+
+* [Andrew Clark's talk on React Suspense](https://www.youtube.com/watch?v=z-6JC0_cOns)
+* [Dan Abramov's talk on React Suspense](https://www.youtube.com/watch?v=6g3g0Q_XVb4) from ReactFest
+* [Jared Palmer's talk on React Suspense](https://www.youtube.com/watch?v=SCQgE4mTnjU) from React Conf 2018
+* [Talk of Andrew Clark and Brian Vaughn on Concurrent Rendering in React](https://www.youtube.com/watch?v=ByBPyMBTzM0) from React Conf 2018
+
+### Articles And Documentation
+
+* [Code Splitting the React app - official documentation]([https://reactjs.org/docs/hooks-intro.html](https://reactjs.org/docs/code-splitting.html))
```

Comment:
wrong syntax - this link does not render in https://github.com/msd-code-academy/react-workshop/blob/5a313446c1528225744202effd9897481905909b/04-suspense-and-async-mode/README.md#articles-and-documentation

Reply:
Thank you, good catch. Blame VS code, it's too smart when copying links to markdown 😂 Fixed in de7327d